src/utils/hooks.ts

```diff
@@ -70,8 +78,15 @@ export function useInput(
 
 export function useDropdown<T>(
   initialValue: T | undefined,
+  deps: DependencyList = [],
 ): [T | undefined, (event: React.ChangeEvent<{ value: unknown }>) => void] {
   const [value, setValue] = useState<T | undefined>(initialValue);
+
+  useEffect(() => {
+    setValue(initialValue);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [initialValue, ...deps]);
```

Comment:
it is 😮 but there doesn't seem to be a way to define a custom hook with dependencies without this? the linter error doesn't like that i'm spreading `deps` because it can't statically check if all the dependencies are in there, but it seems very reasonable to want to surface deps for a custom hook

and stackoverflow's suggestion wasn't very compelling to me: https://stackoverflow.com/a/67802751